### DISCUSSION TITLE
CodeMagic is unable to prepare a build machine.

### DISCUSSION THREAD

**Discussion by casanteyApps** [codemagic-ci-cd/codemagic-docs#2050] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Update remote repository on github
CodeMagic webhook is triggered to automatically build  the application based on the options specified in my codemagic.yaml file. 
CodeMagic is unable to prepare a build machine. (It takes 10 minutes and throws the error below)
<img width="1171" alt="Screenshot 2023-09-16 at 8 32 29 AM" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/118912356/6825eafc-b2a7-4ef5-9558-520170761be2">

I have restarted the build both manually and automatically but the result remains the same!

### Expected results

Build machine should not take longer than usual to be prepared

### Actual results

It takes 10 minutes to prepare and throws an error

### Build id (optional)

6505630aa01b03909d187990

**Comment by dtrdic** [comment]
Hello @casanteyApps, the error logs state that you have `"Invalid encryption key - encrypted variables work only with builds in the same team they were created with"`
Looks like you have moved your app, and you are currently using the same Android keystore. However, using the same keystore across different teams is not allowed. To resolve this, you will need to generate a new .jks file and add it to your workflow.

**Comment by rafeysup93** [comment, marked as answer]
Hi @casanteyApps, 

Please note that you should contact our team through the chat widget in the UI for a quicker response, as your team is on a paid plan.

You might be trying to use the Encrypted Variables in a YAML file with a different team. The build will fail because each team has a unique encryption key.

If you look at your codemagic.yaml file, you will notice that you are using an old way, now deprecated, of using encrypted environment variables in the YAML itself. You can see examples like this:
`Encrypted(stringofvalueshere).`

The encryption tool is now replaced with an easier and less confusing option. Please go to the UI's Environment variables section, create a variable name, and put in its value. Mark the "Secure" checkbox. Create a group name and add it.

It means you no longer need to use env vars in the yaml file. Here is a [doc](https://docs.codemagic.io/variables/environment-variable-groups/) about it.

P.S. To store binary files (keystores, provisioning profiles) as environment variables, you will need to use cat your_file_name.extension |base64 |pbcopy (for macOS )command line first which will copy the result to the clipboard and then go ahead and paste it in the values input and mark the "Secure" checkbox.

On marking the secure checkbox, your file will be encrypted.

**Reply by casanteyApps** [reply]
I will check and revert

**Reply by rafeysup93** [reply]
@casanteyApps, Did you get a chance to check this?
